test/dynamo/test_unspec.py

```diff
@@ -248,6 +248,8 @@ def fn(x, scaler):
         self.assertTrue(same(ref, res))
         self.assertEqual(ref.device, res.device)
 
+    @unittest.expectedFailure
+    # can't turn ToFloat(Proxy(s1)) intermediate size call into compute
```

Comment:
Err, this just seems like a regression right?

Reply:
@ezyang Hmm, what would be the right way to reify the last 2 size calls here?
```
def forward(self, s0 : torch.SymInt, s1 : torch.SymInt, s2 : torch.SymInt, L_image_ : torch.Tensor):
    l_image_ = L_image_
    getitem = l_image_[None]
    interpolate = torch.nn.functional.interpolate(getitem, size = None, scale_factor = 1.873536229133606, mode = 'bilinear', recompute_scale_factor = True, align_corners = False)
    image = interpolate[0]
    size = image.size()
    getitem_2 = size[0]  # shape: s0
    getitem_3 = size[1]  # shape: TruncToInt(1.87353622913361*ToFloat(s1))
    getitem_4 = size[2]  # shape: TruncToInt(1.87353622913361*ToFloat(s1))
    return (getitem_2, getitem_3, getitem_4)
```

I think the issue is maybe with _sympy/interp(), the ToFloat(s1) call gets handled by PythonReferenceAnalysis as a basic Python `float(s1)` call which leads to the error.

Reply:
Ah, make it call torch.sym_float rather than float

Reply:
Ah I see, thank you! That worked, but now I'm seeing a downstream error with TruncToInt, where the input is a Proxy, and the [check](https://github.com/pytorch/pytorch/blob/e9aefad641f24072f36badf4b221ebca3f222966/torch/utils/_sympy/functions.py#L655) `if number in (sympy.oo, int_oo):` tries to go to control flow tracing and fails:
```
  File "/data/users/pianpwk/pytorch/torch/utils/_sympy/functions.py", line 655, in eval
    if number in (sympy.oo, int_oo):
  File "/data/users/pianpwk/pytorch/torch/fx/proxy.py", line 447, in __bool__
    return self.tracer.to_bool(self)
  File "/data/users/pianpwk/pytorch/torch/fx/proxy.py", line 307, in to_bool
    raise TraceError('symbolically traced variables cannot be used as inputs to control flow')
torch._dynamo.exc.InternalTorchDynamoError: symbolically traced variables cannot be used as inputs to control flow
```

The full expression here is `TruncToInt(1.87353622913361*ToFloat(s1))`

Would you know how to change the handler or sympy functions to work well with Proxys?

Reply:
You shouldn't get here with proxies. Everything in PythonReferenceAnalysis must be traceable. If you look further up your call stack you should find out which method needs to be overridden.

Reply:
Oh, I think it's because PythonReferenceAnalysis doesn't implement this, it goes all the way to ReferenceAnalysis.

Reply:
I guess one issue is, when we're converting a shape call on an intermediate tensor into compute on input shapes, we're checking whether all expression symbols are tracked & we're operating any on non-placeholder nodes, but we're unaware if PythonReferenceAnalysis supports reifying the call - this happens with other calls like int_truediv too.

Reply:
Everything is supposed to be supported. Anything that's missing is a bug :o)